modules/spring-config/src/main/java/org/mule/runtime/config/internal/factories/FlowRefFactoryBean.java

```diff
@@ -100,6 +107,8 @@
   @Inject
   private ConfigurationComponentLocator locator;
 
+  private static final Object referencedProcessorLock = new Object();
```

Comment:
this should not be final, because in that case multiple applications in the same runtime may be synchronized with each other needlessly. Better get the lock from the MuleContext (there already are some app level locks there.)

Reply:
I decided to lock on the ApplicationContext instance and it's working, sounds ok for you?